### DISCUSSION TITLE
Hosting payload with nextjs

### DISCUSSION THREAD

**Discussion by SyedMuzamilM** [payloadcms/payload#1167] [Q&A]
Hi everyone.
I want to host nextjs with payload cms on AWS EC2 using nginc server. I just want to know the best way to do that.
I tried to build it and then setup the nginx but I was not able to access the admin panel.

**Comment by falko100** [comment]
You probably did everything right, except for setting up the `serverURL` in the config in `payload.config.ts`.
I have had the same problem.

**Reply by DanRibbens** [reply]
Pro tip for everyone—when serving the API and admin panel from a single domain, you can remove the `serverURL` property altogether and Payload will just work with relative urls.

**Reply by SyedMuzamilM** [reply]
> You probably did everything right, except for setting up the `serverURL` in the config in `payload.config.ts`. I have had the same problem.

Yes I figured it out later. And I moved it to a separate project away  from nextjs.

It worked fine and I hosted it on EC2 as well.

But when I moved payload to my existing nodejs api. I gave me errors when accessing admin page

![Screenshot from 2022-09-29 14-34-17](https://user-images.githubusercontent.com/38078427/192990399-aad1b2a6-db12-4653-9375-835cefdce612.png)
